### PR TITLE
[ARM] az policy assignment list: Support listing policy assignments at Management Group scope

### PR DESCRIPTION
**Pull request by pilor**
This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by pilor** [comment]
Fixes #9067

**Comment by arrownj** [comment]
Hi @zhoxing-ms @Juliehzl , could you please help review this PR ?

**Comment by yonzhan** [comment]
add to S165.

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -640,7 +641,7 @@ def _get_custom_or_builtin_policy(cmd, client, name, subscription=None, manageme
         if cmd.supported_api_version(min_api='2018-03-01'):
             if not management_group:
                 management_group, name = _parse_management_group_reference(name)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py]: In `_parse_management_group_reference()`, when `_is_management_group_scope() == true` and `len(parts) < 9`, the name will be set to null affecting the call to `policy_operations.get(name)` below.

Comment (pilor) [reply src/azure-cli/azure/cli/command_modules/resource/custom.py]: good catch, I'm trying to overload too much logic in this one parse method

**Comment by pilor** [comment]
@zhoxing-ms Bug fixed, added a test case for this and merged from dev

**Pull request merged**